tools/expotools/src/commands/UpdateVersionsEndpoint.ts

```diff
@@ -0,0 +1,148 @@
+
+  const sdkVersionConfig = containsSdk ? { ...versions.sdkVersions[sdkVersion] } : {};
+
+  console.log(`\nUsing ${chalk.blue(host)} host ...`);
+  console.log(`Using SDK ${chalk.cyan(sdkVersion)} ...`);
+
+  if ('deprecated' in options) {
+    setSdkVersionConfig(sdkVersionConfig, 'isDeprecated', !!options.deprecated);
+  }
+  if ('releaseNoteUrl' in options && typeof options.releaseNoteUrl === 'string') {
+    setSdkVersionConfig(sdkVersionConfig, 'releaseNoteUrl', options.releaseNoteUrl);
+  }
+  if (options.key) {
+    if (!('value' in options) && !options.delete) {
+      console.log(chalk.red('`--key` flag requires `--value` or `--delete` flag.'));
+      return;
+    }
+    setSdkVersionConfig(sdkVersionConfig, options.key, options.delete ? undefined : options.value);
+  }
+
+  const newVersions = {
+    ...versions,
+    sdkVersions: {
+      ...versions.sdkVersions,
+      [sdkVersion]: sdkVersionConfig,
+    },
+  };
+
+  const delta = jsondiffpatch.diff(versions.sdkVersions[sdkVersion], sdkVersionConfig);
+
+  if (!delta) {
+    console.log(chalk.yellow('There are no changes to apply in the configuration.'));
+    return;
+  }
+
+  console.log(`\nHere is the diff of changes to apply on SDK ${chalk.cyan(sdkVersion)} version config:`);
+  console.log(
+    jsondiffpatch.formatters.console.format(delta!, versions.sdkVersions[sdkVersion]),
+  );
+
+  const { isCorrect } = await inquirer.prompt<{ isCorrect: boolean }>([
+    {
+      type: 'confirm',
+      name: 'isCorrect',
+      message: `Does this look correct? Type \`y\` or press enter to update ${chalk.green(environment)} config.`,
+      default: true,
+    },
+  ]);
+
+  if (isCorrect) {
+    // Save new configuration.
+    try {
+      await Versions.setVersionsAsync(newVersions);
+    } catch (error) {
+      console.error(error);
+    }
+
+    console.log(
+      chalk.green(`\nSuccessfully updated ${environment} config. You can check it out on`),
+      chalk.blue(`https://${host}/--/api/v2/versions`),
+    );
+  } else {
+    console.log(chalk.yellow('Canceled'));
+  }
+}
+
+export default (program: Command) => {
+  program
+    .command('update-versions-endpoint')
+    .alias('update-versions')
+    .description(`Updates SDK configuration under ${chalk.blue('https://staging.expo.io/--/api/v2/versions')}`)
+    .option('-s, --sdkVersion [string]', 'SDK version to update. Can be chosen from the list if not provided.')
+    .option('-d, --deprecated [boolean]', 'Sets chosen SDK version as deprecated.')
+    .option('-r, --release-note-url [string]', 'URL pointing to the release blog post.')
+    .option('-k, --key [string]', 'A custom, dotted key that you want to set in the configuration.')
+    .option('-v, --value [any]', 'Value for the custom key to be set in the configuration.')
+    .option('--delete', 'Deletes config entry under key specified by `--key` flag.')
+    .option('--production', 'Whether to use production config instead of staging. Use carefully!', false)
```

Comment:
i don't think we should even allow this as an option, for a few reasons: (1) forcing ourselves to add the extra step of making a change on staging and then checking it in the promotion step is a good way to ensure we think carefully about any change we make and avoid mistakes. (2) having it on staging makes it more likely we'll actually test on staging. (3) we want to avoid situations where production gets "ahead" of staging, because then the changes can get erased easily with the `promote` command. let me know what you think 😄

Reply:
These are a lot of really good points, especially the one about making sure production doesn't get ahead of staging.

Reply:
I completely agree with you about not going ahead of staging. The intention of this flag was to provide a second way to update productions config, as I remember `promote-versions-to-prod` didn't work many times and we used to be pushing the whole config to production by different script called `overwrite-versions` in https://github.com/expo/universe/blob/a2e2e207684fa54603499f36a0a331d2df5ae9bf/tools/gulpfile.babel.js#L329
This way would be much more safe than `overwrite-versions`.
Maybe we can add appropriate warning when `--production` is used?

Reply:
in the case where `promote-versions-to-prod` is broken, the correct way to write to production would be to fix `promote-versions-to-prod` 😄 , or if that's not doable for some reason, to overwrite the entire production endpoint with staging (basically doing the same thing as `promote-versions-to-prod`, just manually). I don't think we should ever be setting individual keys directly to prod -- it's just too error-prone.